### DISCUSSION TITLE
Make prompt characters non-copyable in command-line snippets in markdown.

### DISCUSSION THREAD

**Discussion by anderslatif** [community/community#114734] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

I've always wanted to be able to define markdown command line snippets with prompt characters like '>' or '$' to signify that this is something to run in the terminal but doing so makes it harder to copy the command. 

Example:

\```bash
$ echo "hello world"
\```

Should only copy `echo "hello world"`. Same with this:

\```powershell
$ echo "hello world"
\```

**Comment by chuckwondo** [comment]
Agreed. Also, for language-specific REPL prompts, like this in the Python REPL:

```python
>>> print("hello world")
```

I don't want `>>> ` copied to the clipboard.

**Comment by anderslatif** [comment]
It is already supported in Github Packages example commands. When you copy the command the prompt character is omitted. What we need now is a syntax for it:

https://github.com/github/incubator-airflow/pkgs/container/incubator-airflow%2Fpython

**Comment by pevinskevin** [comment]
Yes, GitHub plz fix!

**Comment by LinusSch** [comment]
#### More motivation
A couple of cases where this would be extra valuable:

1. Distinguishing between root prompt and standard prompt:
    ```console
    user@machine:~$ sudo su
    root@machine:/home/user# exit
    ```
2.  Instructing users to use a Python virtual environment, like so:
    ```console
    user@machine:~$ mkdir env
    user@machine:~$ python -m venv env
    user@machine:~$ source env/bin/activatepip install --upgrade pip
    (env) user@machine:~$ pip install -r requirements.txt
    ```

And likewise, any other case where a changing prompt is an important guide for the user (as changing prompts are designed to be).

#### Bonus points
If output can also be shown but not copied, e.g. the following example having only `sudo su` and `exit` as copyable characters:
```console
user@machine:~$ sudo su
[sudo] password for user:
root@machine:/home/user# exit
```

**Reply by anderslatif** [reply]
Being able to add entirely custom non-copyable characters would be a treat. Great suggestions!

**Comment by tjoneseng** [comment]
Duplicate of #35615

**Reply by yorkshirelandscape** [reply]
> Duplicate of #35615

It's a little hard to understand why this very commonplace feature continues to be unavailable in GitHub years after it was first suggested.

**Reply by anderslatif** [reply]
I actually think that @LinusSch suggestions make this a unique feature request compared to supporting `console` and only removing one prompt character.

**Comment by bikstok** [comment]
fix now github pls
